app/components/answer-tile/custom-extra.js

```diff
@@ -0,0 +1,90 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  store: Ember.inject.service(),
+  child_question: null,
+  child_question_tag: null,
+  didReceiveAttrs() {
+    var member = this.get('member'),
+      chapter = this.get('chapter'),
+      parent_question = this.get('question'),
+
+    // The parent question must have a child question to use the custom-extra component
+      child_question = parent_question.get('questions').objectAt(0);
+
+    // Find the tag corresponding to the child question
+    let tags = member.get('tags')
+      .filterBy('chapterId', parseInt(chapter.id))
+      .filterBy('questionId', parseInt(child_question.id)),
+      tag;
+
+    // Find the corresponding tag, or create it if it doesn't exist
+    if (tags.get('length')) {
+
+      // The tag exists, so use that one
+      tag = tags.objectAt(0);
+    } else {
+
+      // The tag does not exist yet, so create it
+      tag = this.get('store').createRecord('tag', {
+        member: member,
+        chapterId: parseInt(chapter.id),
+        questionId: parseInt(child_question.id),
+        answer: [],
+      });
+    }
+
+    // Reorder options so that input is before the special options
+    let options = parent_question.get('options'),
+      index_input,
+      index_first_special_option = null,
+      option_input;
+
+    options.forEach((option, i) => {
+      if (option.get('text').indexOf('__input') === 0) {
+        index_input = i; // Mark the position of the input option
+        option_input = option; // Create reference to the input option
```

Comment:
Remove the option from the list here. Removing outside of this conditional could error because there may not be a valid index to remove at.